### PR TITLE
Update doc regarding clustering and node management [skip ci]

### PR DESCRIPTION
**Pull request by michalwski**
From MongooseIM 2.0.0 there were quite significant changes to clustering.
I removed doc for older version in order to make it more understandable, doc for older version is still here in git repo and in readthedocs.io when a prev version is chosen.

### CODE REVIEW

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md [hunk 1]
```diff
@@ -21,24 +21,24 @@ mongooseim   hard   nofile   1000000
 
 If you are using **Ubuntu**, all `/etc/pam.d/common-session*` files should include `session required pam_limits.so`.
 
-### `vm.args` and `vars.config` files
+### `vm.args` file
 
-Before making a release, some values in `rel/files/vm.args` have to be changed. 
-When using an SSL/TLS connection we advise to increase `ERL_MAX_PORTS` to `350000`. 
+This file is located in `REL_ROOT/etc/vm.args` where `REL_ROOT` is the path
```
Comment (goddammit) [inline doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: This file is located in `REL_ROOT/etc/vm.args` where `REL_ROOT` is the path to a MonoogseIM release (ie. `_build/prod/rel/mongooseim` if you build MongooseIM from source).

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md [hunk 2]
```diff
@@ -21,24 +21,24 @@ mongooseim   hard   nofile   1000000
...
+to MonoogseIM release (ie. `_build/prod/rel/mongooseim` if you build MongooseIM from source).
+This file contains erlang options used when starting the VM.
+
+When using an SSL/TLS connection we advise to increase `ERL_MAX_PORTS` to `350000`.
 This value specifies how many ports (files, drivers, sockets etc) can be used by Erlang VM.
-Be cautious - it preallocates some structures inside the VM and will have impact on memory usage. 
+Be cautious - it preallocates some structures inside the VM and will have impact on memory usage.
```
Comment (goddammit) [inline doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: Be cautious - it preallocates some structures inside the VM and will have impact on the memory usage.

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md [hunk 3]
```diff
@@ -53,71 +53,51 @@ Make sure that the range you set provides enough ports for all the nodes in the
...
-### Other
-
-The approaches described above can be mixed - we can use DNS load balancing to pick a software load balancer which will select one of the nodes.
-
 ## Connecting nodes
 
 Checklist:
 
-- working directory `rel/mongooseim`
+- working directory `rel/mongooseim` (root of MongooseIM release or installation)
```
Comment (goddammit) [inline doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: working directory `rel/mongooseim` (root of a MongooseIM release or installation)

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md [hunk 4]
```diff
@@ -53,71 +53,51 @@ Make sure that the range you set provides enough ports for all the nodes in the
...
+If you type `yes` MongooseIM will start joining the cluster.
+Successful output may look like the following:
 
-For an older `MongooseIM` please see the previous versions of the documentation.
+```
+You have successfully joined the node mongooseim2@localhost to the cluster with node member mongooseim@localhost
+```
 
-### Remove node
+In order to skip the question you can use `-f` which will perform the action without any warning and interaction.
```
Comment (goddammit) [inline doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: In order to skip the question you can use `-f` which will perform the action without any warning or interaction?

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md [hunk 5]
```diff
@@ -126,32 +106,37 @@ mongooseimctl leave_cluster
 
 It only makes sense to use it if the node is the part of a cluster, e.g `join_cluster` was called from that node before.
 
-The successful output from the above command starts with `You have successfully left the node from the cluster`.
+Similarly to `join_cluster` a warning and a question will be displayed unless option `-f` is added to the command.
```
Comment (goddammit) [inline doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: Similarly to `join_cluster` a warning and a question will be displayed unless the option `-f` is added to the command.

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md [hunk 6]
```diff
@@ -126,32 +106,37 @@ mongooseimctl leave_cluster
...
 
 ```bash
 mongooseimctl remove_from_cluster RemoteNodeName
 ```
 
-where `RemoteNodeName` is a name of the node that we'd like to remove from our cluster. 
+where `RemoteNodeName` is a name of the node that we'd like to remove from our cluster.
 This command could be useful when the node is dead and not responding and we'd like to remove it remotely.
-The successful output from above command starts with `The node has been removed from the cluster`
+The successful output from above command may look like the following:
```
Comment (goddammit) [inline doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: The successful output from the above command may look like the following:

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md [hunk 7]
```diff
@@ -126,32 +106,37 @@ mongooseimctl leave_cluster
...
 
-#### MongooseIM before 1.7.0
-
-For an older `MongooseIM` please see the previous versions of the documentation.
+`The node mongooseim2@localhost has been removed from the cluster`
 
 ### Cluster status
 
-To see if a newly added node is properly clustered or to see how the cluster looks like, the following commands can be useful (on any running node in the cluster).
+To see if a newly added node is properly clustered or to see how the cluster looks like,
```
Comment (goddammit) [inline doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: You can use the following commands on any of the running nodes to examine the cluster or to see if a newly added node is properly clustered:

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md [hunk 8]
```diff
@@ -162,5 +147,31 @@ To see stopped or misbehaving nodes following command can be useful:
...
 
-This command shows which nodes are considered stopped. 
+This command shows which nodes are considered stopped.
 This does not necessarily indicate that they are down but might be a symptom of a communication problem.
+
+## Load Balancing
+
+### Elastic Load Balancer (ELB)
+
+When using ELB pleas be advised that some worm-up time maybe needed before
```
Comment (goddammit) [inline doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: When using ELB please be advised that some warm-up time may be needed before the load balancer works efficiently for a big load.

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md (hunk 4 above)
Comment (michalwski) [reply doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: How about:

In order to skip the question you can add option `-f` which will perform the action without displaying the warning and waiting for confirmation

Does it sound better @goddammit ?

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md (hunk 7 above)
Comment (michalwski) [reply doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: Brilliant!

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md (hunk 4 above)
Comment (goddammit) [reply doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: In order to skip the question you can add option -f which will perform the action without displaying the warning and waiting for the confirmation.

:)

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md [hunk 9]
```diff
@@ -2,15 +2,15 @@
 
 ### File descriptors
 
-To handle large traffic, some of the system variables need to be tuned. 
-Number one on that list is the maximum number of file descriptors which by default is set to 1024. 
+To handle large traffic, some of the system variables need to be tuned.
+Number one on that list is the maximum number of file descriptors which by default is set to 1024.
```
Comment (Nyco) [inline doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: Does this consider both systemd and non-systemd systems?

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md [hunk 10]
```diff
@@ -21,24 +21,24 @@ mongooseim   hard   nofile   1000000
...
 If you are using **Ubuntu**, all `/etc/pam.d/common-session*` files should include `session required pam_limits.so`.
 
-### `vm.args` and `vars.config` files
+### `vm.args` file
 
-Before making a release, some values in `rel/files/vm.args` have to be changed. 
-When using an SSL/TLS connection we advise to increase `ERL_MAX_PORTS` to `350000`. 
+This file is located in `REL_ROOT/etc/vm.args` where `REL_ROOT` is the path to a MonoogseIM release
+(ie. `_build/prod/rel/mongooseim` if you build MongooseIM from source).
+This file contains erlang options used when starting the VM.
```
Comment (Nyco) [inline doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: Maybe we should start this section by this sentence.
=> First we explain what that file is for, and then where it is located (most important first?).
So a swap of lines 26 and 28.

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md (hunk 9 above)
Comment (fenek) [reply doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: Indeed I would rather say "which e.g. in many Linux systems is set to 2014."

File: doc/operation-and-maintenance/Cluster-configuration-and-node-management.md [hunk 11]
```diff
@@ -21,24 +21,24 @@ mongooseim   hard   nofile   1000000
 
 If you are using **Ubuntu**, all `/etc/pam.d/common-session*` files should include `session required pam_limits.so`.
```
Comment (fenek) [inline doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: Why?

Comment (michalwski) [reply doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: Why not? This line I didn't touch and from what I remember this worked when the page was first created.

Comment (michalwski) [reply doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: Typically there are `common-session` and `common-session-noninteractive`. Both should have the mentioned line if we want to be sure that the limits will be changed. If someone knows only one will be best for it then I assume this is quite experienced user who knows his system and will set it as he needs.
As you can see above the limits are only for user `mongooseim` so other users will connect with their limits anyway.

Comment (fenek) [reply doc/operation-and-maintenance/Cluster-configuration-and-node-management.md]: Sorry, for a moment I was sure it was related to PAM auth backend that we've removed long ago. :)

**Pull request merged**